piperider_cli/runner.py

```diff
@@ -548,6 +548,26 @@ def _check_test_status(assertion_results, assertion_exceptions, dbt_test_results
     return True
 
 
+def check_dbt_manifest_compatibility(ds: DataSource, dbt_state_dir: str):
+    database = ds.get_database()
+    schema = ds.get_schema()
+
+    manifest = dbtutil._get_state_manifest(dbt_state_dir)
+    models = [x for x in manifest.get('nodes').values() if x.get('resource_type') in ['model']]
+
+    # all model’s schema should be the prefix of target’s schema
+    def filter_schema(model):
+        model_schema = model.get('schema', '')
+        return model_schema == schema or model_schema.startswith(f'{schema}_')
+    filtered_models = list(filter(filter_schema, models))
```

Comment:
this code neglects dbt-labs work on schema and database customization using macros. It enforces to start schema with target name. Instead of using different schemas for production and development environments, we use different GCP projects.